### PR TITLE
Documented PyJWT dependency and improved logging and exception messages

### PR DESCRIPTION
**Pull request by hoylen**
Updated documentation so it is clear that PyJWT (and not any jwt library) is needed. Corrected "pip install" command to install "pyjwt" instead of "jwt".

Added more information to the `NotImplemented` exception messages. Instead of saying "Subclasses must implement this function", it is now more useful because it says "Missing function implementation in className: nameOfMissingFunction".

Extra logging has been added to help in debugging. During development, the _signature base string_ signed by the client may be different from the one independently calculated by the server: logging these string helps debug such problems.

In _oauthlib/oauth1/rfc5849/endpoints/signature_only.py_ exceptions are caught and then False is simply returned. The details in those exceptions are useful for debugging what went wrong. Their values are now logged, rather than simply discarded.

### CODE REVIEW

File: docs/oauth1/server.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Creating a Provider
 ===================
 
-OAuthLib is a dependency free library that may be used with any web
+OAuthLib is a framework independent library that may be used with any web
```
Comment (thedrow) [inline docs/oauth1/server.rst]: Why did you change this?

Comment (hoylen) [inline docs/oauth1/server.rst]: Because OAuthLib has dependencies, so it is not "dependency free". The description of "dependency free" would mean it does not need any other package to run. OAuthLib (as indicated in the _requirements.txt_ file) needs PyJWT, blinker and cryptography packages to function.

I'm trying to guess what the original author really meant, and assume they meant to say it can be used with any framework (e.g. doesn't have to be Django) rather than "dependency free" (which it is clearly not, since it has dependencies).

**Comment by thedrow** [comment]
Can you please rebase this branch?

**Comment by thedrow** [comment]
Looks like the rebase went really wrong here. Can you try again?

**Comment by hoylen** [comment]
Are you referring to the red cross next to 13387a4? There was a bug in my code, so it was failing the automatic checks. That was then fixed in b3f4cb9.

**Comment by thedrow** [comment]
I'm referring to the entire code history in this PR that isn't related. See https://github.com/idan/oauthlib/pull/458#commits-pushed-bf2f9db

**Comment by hoylen** [comment]
Done. This should be better.

**Comment by thedrow** [comment]
Thanks!
